cli/src/opts/multi_wallet.rs

```diff
@@ -287,12 +291,20 @@ impl MultiWallet {
     }
 
     pub fn mnemonics(&self) -> Result<Option<Vec<LocalWallet>>> {
-        if let (Some(mnemonic_paths), Some(mnemonic_indexes)) =
-            (self.mnemonic_paths.as_ref(), self.mnemonic_indexes.as_ref())
-        {
+        if let Some(mnemonic_paths) = self.mnemonic_paths.as_ref() {
             let mut wallets = vec![];
-            for (path, mnemonic_index) in mnemonic_paths.iter().zip(mnemonic_indexes) {
-                wallets.push(self.get_from_mnemonic(path, *mnemonic_index)?)
+            let hd_paths: Vec<_> = if let Some(ref hd_paths) = self.hd_paths {
+                hd_paths.into_iter().map(String::as_str).map(Some).collect()
+            } else {
+                repeat(None).take(mnemonic_paths.len()).collect()
+            };
+            let mnemonic_indexes: Vec<_> = if let Some(ref mnemonic_indexes) = self.mnemonic_indexes {
+                mnemonic_indexes.into_iter().copied().collect()
```

Comment:
Not sure if there is a shorter way to do that?

Reply:
could also do a `map` + `unwrap_or_else`, but this is more readable